doc/nrf/libraries/modem/nrf_modem_lib.rst

```diff
@@ -75,6 +75,12 @@ The module provides the functionality for starting, stopping, and forwarding of
 
 If the application wants the trace data, :c:func:`nrf_modem_lib_trace_init` must be called before :c:func:`nrf_modem_lib_init`.
 This is done automatically when using the OS Abstraction layer.
+
+Trace processing is done in the ``trace_handler_thread``.
+When the modem trace module receives trace data, it places it in a FIFO queue.
+The thread reads from the FIFO queue and forwards the trace data to the configured trace transport medium.
+Since FIFO queues use heap memory, the use of modem trace module results in increased heap usage.
```

Comment:
We have now a dedicated heap, so strictly speaking it does not apply to the (system) heap anymore.
Do we want to reference its KConfig option here?

Reply:
Updated the docs. Can you please have a look at the change @umapraseeda ?